src/ngraph/runtime/cpu/quantization_util.hpp

```diff
@@ -0,0 +1,81 @@
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#pragma once
+#include <vector>
+#include "ngraph/node.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/runtime/cpu/op/quantize.hpp"
+#include "ngraph/util.hpp"
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace cpu
+        {
+            namespace quantization_util
+            {
+                static inline void get_min_max_range(const ngraph::Node* node,
+                                                     std::vector<float>& quant_util)
+                {
+                    auto quantize = static_cast<const ngraph::op::Quantize*>(node);
+                    auto min_const_op =
+                        std::dynamic_pointer_cast<ngraph::op::Constant>(quantize->get_argument(1));
+                    auto max_const_op =
+                        std::dynamic_pointer_cast<ngraph::op::Constant>(quantize->get_argument(2));
+                    float input_min_range =
+                        *(static_cast<float const*>(min_const_op->get_data_ptr()));
+                    float input_max_range =
+                        *(static_cast<float const*>(max_const_op->get_data_ptr()));
+                    // begin code copied and pasted from
+                    // github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/quantize_op.cc
+                    float min_range;
+                    float max_range;
+                    // If input_min_range and input_max_range are close,
+                    // introduce a slightly larger delta between them.
+                    min_range = std::min(0.0f, input_min_range);
+                    const float epsilon =
+                        std::max(1.0f, std::max(fabsf(input_min_range), fabsf(input_max_range))) /
+                        100.0f;
+                    max_range = std::max(input_max_range, min_range + epsilon);
+                    max_range = std::max(0.0f, max_range);
+                    // end code copied and pasted from
+                    // github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/quantize_op.cc
+                    const float max_abs = std::max(std::abs(min_range), std::abs(max_range));
+                    bool is_signed = (quantize->get_quantize_et()).is_signed();
+                    const float target_range =
+                        static_cast<float>((is_signed ? std::pow(2, 7) : std::pow(2, 8)) - 1);
+                    max_range = max_abs;
+                    min_range = is_signed ? -max_abs : 0;
+                    const float scale = target_range / max_abs;
```

Comment:
I am not convinced that a quantize API consisting of min/max/type is sufficient.  Take a look at the TF quantize op here:

https://www.tensorflow.org/api_docs/python/tf/quantize

There are three ways to calculate 'scale' from min_range and max_range corresponding to the three 'mode' attributes in tf.quantize (MIN_COMBINED, MIN_FIRST, SCALED).

How can this single method of computing scale map to three methods of computing scale at the framework level?  NOTE:  There are three modes *just* within TF, not to mention other frameworks.

I am thinking about the design of the nGraph core quantize (and dequantize) op.  My initial guess was to force the framework to provide the 'scale' in the quantize API.

Reply:
We'd have to use Eigen for the other modes.

Reply:
Add scale and zero-point args. I think with all four, a bridge can deal with all the modes.
So Q = max(R-min, max-min)/scale+zero_point

Reply:
@diyessi does it have to go in this PR or we'd be adding it for the ngraph quantize op?

Reply:
We can add it later.